lumicks/pylake/correlated_stack.py

```diff
@@ -46,9 +99,21 @@ def __init__(self, tiff_file):
     def get_frame(self, frame):
         return TiffFrame(self._src.pages[frame])
 
+    def align(self):
+        """ reconstruct all pages using alignment matrices from Bluelake 
+            return as new TiffStack
+        """
+        buffer = io.BytesIO()
+        with tifffile.TiffWriter(buffer) as writer:
+            for j in range(self.num_frames):
+                frame = self.get_frame(j)
+                aligned_img = frame.align()
+                writer.save(aligned_img)
+        buffer.seek(0)
```

Comment:
While I do like keeping the internals very much the same, and providing a TIFF image, there's a fairly critical issue with doing it this way; namely that other TIFF properties (such as tags) get lost.

In the current implementation the metadata gets lost (e.g. `.start` and `.stop` no longer work post alignment). Note that there are a few tests regarding correlated stacks (checking `start`, `stop` etc), and it would be good to see whether these pass post alignment as well. If you do it like this, it would be good to add a test for this (or rather, somehow parameterize the existing `CorrelatedStack` test to optionally include an alignment procedure so that future additions to `CorrelatedStack` also immediately test the aligned stacks to see whether all the attributes still work). 

Alternatively, I wonder whether it isn't preferable to lazily perform this alignment when the data is requested, rather than having a function which returns a copy of the whole TIFF stack with modified frames. Perhaps the frame `.data` can return an aligned image when alignment information is available and we add a field named `raw_data` or `unaligned_data` which doesn't apply the processing. As for plotting, there could be a named keyword `aligned` or something. I would assume that in most cases, users want the aligned images.

Thoughts @facutuesca @dean0x7d ?

Reply:
Addendum: I discussed it briefly with Roeland, and the aligned image should be the default; which kinda makes me think the lazy evaluation with aligned as default makes more sense.